homeassistant/components/switch/rpi_rf.py

```diff
@@ -101,22 +105,23 @@ def is_on(self):
         """Return true if device is on."""
         return self._state
 
-    def _send_code(self, code, protocol, pulselength):
-        """Send the code with a specified pulselength."""
-        _LOGGER.info("Sending code: %s", code)
-        res = self._rfdevice.tx_code(code, protocol, pulselength)
-        if not res:
-            _LOGGER.error("Sending code %s failed", code)
-        return res
+    def _send_code(self, code_list, protocol, pulselength):
+        """Send the code(s) with a specified pulselength."""
+        _LOGGER.info("Sending code(s): %s", code_list)
+        for code in code_list:
+            self._rfdevice.tx_code(code, protocol, pulselength)
+        return True
 
     def turn_on(self):
         """Turn the switch on."""
-        if self._send_code(self._code_on, self._protocol, self._pulselength):
+        if self._send_code(self._code_on_list, self._protocol,
```

Comment:
If you're going to hard-code the `return True` above in `_send_code`, do you really need to check these conditionals?